protostar/cheatable_starknet/cheatcodes/cairo_cheatcode.py

```diff
@@ -1,12 +1,65 @@
-from abc import ABC
-from typing import TYPE_CHECKING
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Callable, Union
+
+from starknet_py.cairo.felt import encode_shortstring
 
 from protostar.cairo import HintLocal
+from protostar.cheatable_starknet.cheatable_exception import CheatableException
 
 if TYPE_CHECKING:
     from protostar.cheatable_starknet.cheaters import CairoCheaters
 
+Felt = int
+
+
+@dataclass
+class CheatcodeException(Exception):
+    code: Felt
+    ex: Exception
+
+
+@dataclass
+class CairoCheatcodeInvalidExecution:
+    ok = None
+    err: CheatcodeException
+
+    def unwrap(self) -> Any:
+        raise self.err
+
+
+@dataclass
+class CairoCheatcodeValidExecution:
+    ok: Any
+    err = None
+
+    def unwrap(self) -> Any:
+        return self.ok
+
+
+CairoCheatcodeExecutionResult = Union[
+    CairoCheatcodeValidExecution, CairoCheatcodeInvalidExecution
+]
+
 
 class CairoCheatcode(HintLocal, ABC):
     def __init__(self, cheaters: "CairoCheaters"):
         self.cheaters = cheaters
+
+    @abstractmethod
+    def _build(self) -> Callable:
+        ...
```

Comment:
maybe we should just call this differently? `get_implementation` ?

Reply:
I am not sure if `get_implemention` improves the readability significantly. It should be a private method.

Reply:
Both are fine imo

Reply:
Preserved the original name.